### ISSUE TITLE
Incorrect template errors with union types

### ISSUE THREAD

**Issue by chuckjaz** [angular/angular#22125]
## I'm submitting a...

<pre><code>
[x] Bug report  
</code></pre>

## Current behavior

Imported from https://github.com/angular/vscode-ng-language-service/issues/231

With the following component
```
@Component({
  selector: 'my-app',
  template: `
    <div>
      <h2>Hello {{name}}</h2>
      <p *ngIf="name == 'steve'">hi steve</p>
    </div>
  `,
})
export class App {
  name: string | number;
  constructor() {
    this.name = `Angular!`
  }
}
```

An error is given for the `*ngIf` value: `[Angular] Expected the operants to be of similar type or any`.

## Expected behavior

No template error
appears similar to https://github.com/angular/vscode-ng-language-service/issues/16

**Comment by tomwanzek** [comment]
Having this issue with nullable types and `strictNullChecks` activated.

**Comment by tomwanzek** [comment]
Also, to clarify, the issue also appears in a property binding condition like `[disabled]="id === null"`, where `id : string | null`

**Comment by sdurnov** [comment]
Experiencing the same thing. Pls, fix.

**Comment by PachowStudios** [comment]
Any update on this? Extremely annoying...

**Comment by sovushka-utrom** [comment]
Any update? Error is still here.

**Comment by joshuwhi** [comment]
Error happens with plain string values as well.

``` typescript
mystring: string;

[disabled]="mysting !== 'somestring'"
```
Error:
property *mystring* of *MyComponent*
Expected the operants to be of similar type or any

**Comment by YogurtTheHorse** [comment]
+1 for that one

**Comment by Fourie-r** [comment]
+1 here

**Comment by fjc2112** [comment]
+1, pls fix

**Comment by aerojeyenth** [comment]
This error occurs specifically in [src] and *ngIf bindings. +1.

**Comment by Sathishchary** [comment]
yes, I am also getitng for the ngIf conditions `*ngIf="orderValue === 'desc'" `How to fix that error?

**Comment by joshuwhi** [comment]
I no longer see the issue with the latest language-service plugin on VSCode and after upgrading to angular 8.1.0

**Comment by simeyla** [comment]
Amazed I haven't run into this before.

Temporary fix in my `.ts` file

    notNull = (value: number | null) => value != null;

Then in the template:

     *ngIf="notNull(item.quantities.oneTimeQty)" 

If this is ever fixed it'll be easy to go back and find.

**Comment by ericsnap** [comment]
@simeyla yeah, that's how you _workaround_ it, but remember that expressions like that get evaluated at _every_ change detection. So, if you have hundreds or thousands of items, this could be an issue. We still need a long term solution (ivy?)

**Comment by JoostK** [comment]
This was fixed in #36529 and also won't be an issue anymore with the Ivy language service.
